Docker Desktop 4.15.0 release notes (#16233)
docker/docs: desktop/release-notes.md

For frequently asked questions about Docker Desktop releases, see [FAQs](faqs/general.md/#releases)

## 4.15.0

2022-12-01

> Download Docker Desktop
>
> {% include desktop-install.html %}

### New

> 

Reviewer: ```suggestion

- Substantial performance improvements for macOS users with the option of enabling the new VirtioFS file sharing technology. Available for macOS 12.5 and above. 
- Docker Desktop for Mac does no longer need to install the privileged helper process `com.docker.vmnetd` on install or on the first run. For more information see [Permission requirementsfor Mac](https://docs.docker.com/desktop/mac/permission-requirements/).
```

- Added webassembly capabilities, use with the beta containerd integration.
- Containers will be restored on restart with the containerd integration.
- Fixed listing multi-platform images with the containerd integration.
- Better handling of dangling images with the containerd integration.
- Implement "reference" filter for images with the containerd integration.
- Add support for selecting upstream HTTP/HTTPS proxies automatically via `proxy.pac` in containers, `docker pull` etc.

> - Support custom names for new Dev Environments.

Reviewer 2: ```suggestion
- Docker now supports custom names for new dev environments.
```

> - Updated the beta features settings to better describe experimental features and how people can access them.

Reviewer 2: ```suggestion
- Improved the descriptions for beta and experimental settings to clearly explain the differences and how people can access them.
```

> - Show available disk space of VM in footer.

Reviewer 2: ```suggestion
- Available disk space of VM now displays in the footer of Docker Dashboard.
```

> - Show disk space warning in footer if available space is below 3GB.

Reviewer 2: ```suggestion
- A disk space warning now displays in the footer if available space is below 3GB.
```

> - Support creating Dev Environments by cloning a Git repository to a local bind mount.

Reviewer 2: ```suggestion
- A dev environment can now be created by cloning a Git repository to a local bind mount.
```

- Updated colours and typography to be in line with Docker Design System.

> - Added a Build tab inside Extensions containing all necessary resources to build an extension.

Reviewer 2: ```suggestion
- Added a **Build** tab inside **Extensions** which contains all the necessary resources to build an extension.
```

> - Added the ability to share extensions more easily, either with `docker extension share` CLI or the share button in extensions "Manage" tab.

Reviewer 2: ```suggestion
- Added the ability to share extensions more easily, either with `docker extension share` CLI or with the share button in the extensions **Manage** tab.
```

> - Display number of installs in the extensions marketplace, and ability to sort by number of installs.

Reviewer 2: ```suggestion
- Extensions in the Marketplace now display the number of installs. You can also sort extensions by the number of installs.
```

> - Adds button to open DevEnv folder from UI.

Reviewer 2: Is this meant to be about the browser extension @hyu ?

Reviewer 3: No, this refers to a [change](https://github.com/docker/pinata/pull/19709) where the "open folder" button for Dev Envs works more consistently.

I mentioned the Browser Extension in the [blog](https://docs.google.com/document/d/1Qn-_xfGMl96UqGuh6szdv1ZTd3-NG74gjX0DBKtFjqo/edit#heading=h.nm1ak21mhyx1), but not in these release notes since it's not strictly a part of the DD 4.15 release.

### Upgrades

- [Compose v2.13.0](https://github.com/docker/compose/releases/tag/v2.13.0)
- [Containerd v1.6.10](https://github.com/containerd/containerd/releases/tag/v1.6.10)
- [Docker Hub Tool v0.4.5](https://github.com/docker/hub-tool/releases/tag/v0.4.5)
- [Docker Scan v0.22.0](https://github.com/docker/scan-cli-plugin/releases/tag/v0.22.0)

> ### Bug fixes and minor changes

Reviewer 2: ```suggestion
### Bug fixes and enhancements
```

#### For all platforms

- Fixed regressions when parsing image references on pull. Fixes [docker/for-win#13053](https://github.com/docker/for-win/issues/13053), [docker/for-mac#6560](https://github.com/docker/for-mac/issues/6560), and [docker/for-mac#6540](https://github.com/docker/for-mac/issues/6540).

#### For Mac

> - Improve performance of `docker pull`.

Reviewer 2: ```suggestion
- Improved the performance of `docker pull`.
```

#### For Windows

- Fixed an issue where the system HTTP proxies were not used when Docker starts and the developer logs in.

> - When Docker Desktop is using "system" proxies and if the Windows settings change, Docker Desktop will use the new Windows settings without a restart.

Reviewer 2: ```suggestion
- When Docker Desktop is using "system" proxies and if the Windows settings change, Docker Desktop now uses the new Windows settings without a restart.
```

#### For Linux

> - Fixes hot-reload issue on Linux. Fixes [docker/desktop-linux#30](https://github.com/docker/desktop-linux/issues/30).

Reviewer 2: ```suggestion
- Fixed hot-reload issue on Linux. Fixes [docker/desktop-linux#30](https://github.com/docker/desktop-linux/issues/30).
```

> - Disable tray icon animations on linux, which fixes crashes for some users.

Reviewer 2: ```suggestion
- Disabled tray icon animations on Linux which fixes crashes for some users.
```

For frequently asked questions about Docker Desktop releases, see [FAQs](faqs/general.md/#releases)

## 4.15.0

> {% include release-date.html date="2022-12-01" %}

Reviewer 2: @maxcleme the release-date.html page needs to be updated with the latest download packages and checksums!

Also, the individual buttons on the Linux install packages will need changing as well

Author: @aevesdocker Yes, I'm waiting for the final RC right now, it should updated as soon as I get everything lined up. I'll keep this thread opened until it's fixed. Thanks for the reminder!

Author: Should be good!

> Download Docker Desktop
>
> {% include desktop-install.html %}

### New

- Substantial performance improvements for macOS users with the option of enabling the new VirtioFS file sharing technology. Available for macOS 12.5 and above. 
- Docker Desktop for Mac no longer needs to install the privileged helper process `com.docker.vmnetd` on install or on the first run. For more information see [Permission requirements for Mac](https://docs.docker.com/desktop/mac/permission-requirements/).
- Added [WebAssembly capabilities](wasm/index.md). Use with the [containerd integration](containerd/index.md).
- Improved the descriptions for beta and experimental settings to clearly explain the differences and how people can access them.

> - Available disk space of VM now displays in the footer of Docker Dashboard.

Reviewer 4: This is only available for mac and linux. Should that info be added?

Author: If we look at the footer notes for 4.9.0

https://docs.docker.com/desktop/release-notes/#490

We didn't mentioned it was not fully working on some platforms (eg: it's not showing stats for Windows containers), so I guess it's fine keeping it like this.

@aevesdocker any recommendation maybe?

Reviewer 2: I don't think it would hurt to add 'for Mac and Linux' at the end of the sentence

Author: Added!

- A disk space warning now displays in the footer if available space is below 3GB.

> - Updated colours and typography to be in line with Docker Design System.

Reviewer 4: Do we think people know what the Docker Design System is? Might it be better to instead state the purpose of the changes? Eg increased contrast to meet certain accessibility standards (a11y).

cc @rebecapereira-docker

Reviewer 5: @joe0BAB maybe we can use what we did for the Updates Highlights instead:
`Changes to Docker Desktop's interface as we become more ADA accessible and visually unified.`

> - Updated colours and typography to be in line with Docker Design System.

Reviewer 5: ```suggestion
- Changes to Docker Desktop's interface as we become more ADA accessible and visually unified.
```

For frequently asked questions about Docker Desktop releases, see [FAQs](faqs/general.md/#releases)

## 4.15.0

{% include release-date.html date="2022-12-01" %}

> Download Docker Desktop
>
> {% include desktop-install.html %}

### New

- Substantial performance improvements for macOS users with the option of enabling the new VirtioFS file sharing technology. Available for macOS 12.5 and above. 
- Docker Desktop for Mac no longer needs to install the privileged helper process `com.docker.vmnetd` on install or on the first run. For more information see [Permission requirements for Mac](https://docs.docker.com/desktop/mac/permission-requirements/).
- Added [WebAssembly capabilities](wasm/index.md). Use with the [containerd integration](containerd/index.md).

> - Docker now supports custom names for new dev environments.
> - Improved the descriptions for beta and experimental settings to clearly explain the differences and how people can access them.
> - Available disk space of VM now displays in the footer of Docker Dashboard.
> - A disk space warning now displays in the footer if available space is below 3GB.
> - A dev environment can now be created by cloning a Git repository to a local bind mount.
> - Updated colours and typography to be in line with Docker Design System.
> - Added a **Build** tab inside **Extensions** which contains all the necessary resources to build an extension.
> - Added the ability to share extensions more easily, either with `docker extension share` CLI or with the share button in the extensions **Manage** tab.
> - Extensions in the Marketplace now display the number of installs. You can also sort extensions by the number of installs.
> - Adds button to open DevEnv folder from UI.

Reviewer 3: ```suggestion
- Improved the descriptions for beta and experimental settings to clearly explain the differences and how people can access them.
- Available disk space of VM now displays in the footer of Docker Dashboard.
- A disk space warning now displays in the footer if available space is below 3GB.
- Updated colours and typography to be in line with Docker Design System.
- Added a **Build** tab inside **Extensions** which contains all the necessary resources to build an extension.
- Added the ability to share extensions more easily, either with `docker extension share` CLI or with the share button in the extensions **Manage** tab.
- Extensions in the Marketplace now display the number of installs. You can also sort extensions by the number of installs.
- Dev Environments allow cloning a Git repository to a local bind mount, so you can use any local editor or IDE.
- More Dev Environments improvements: custom names, better private repo support, improved port handling.
```

I collected the lines about Dev Environments together. Decided not to call out the "open folder" button since that feature already exists, we just handle more cases now.
